graphein/ml/datasets/pdb_data.py

```diff
@@ -1789,18 +1789,18 @@ def select_pdb_by_criterion(
             the PDB.
         :type field_values: List[Any]
 
-        :return: The filtered PDB object.
-        :rtype: PandasPdb
+        :return: The filtered PDB object or instead `None` to signify
+        that no atoms within the PDB object were found after filtering.
+        :rtype: Optional[PandasPdb], optional
         """
         for key in pdb.df:
             if field in pdb.df[key]:
                 filtered_pdb = pdb.df[key][
                     pdb.df[key][field].isin(field_values)
                 ]
-                if "ATOM" in key:
-                    assert (
-                        len(filtered_pdb) > 0
-                    ), "Filtered DataFrame must contain atoms."
+                if "ATOM" in key and len(filtered_pdb) == 0:
```

Comment:
I may be being a little pedantic here but I see two possible edge cases with this:

1. Sometimes protein atoms are stored as HETATMs (typically modified residues but this kind of bad practice does happen as an abuse of the PDB format to suit some niche way to store structure data)
2. Similarly, what if the desired selection is actually the HETATM data? Protein-nucleic acid complexes or protein-peptide complexes may store the ligand as a HETATM

Reply:
No problem. Points 1 and 2 didn't come to my mind when I originally implemented this, and I kept expanding on it until now. In light of these points, I think it makes more sense to avoid skipping such DataFrames altogether. It will then be the user's responsibility to "vet" the exported PDB files for these kinds of edge cases in their selected PDBs.

Reply:
I've modified this logic to only issue a warning in such an edge case now.